pkg/chunk/cache/redis_client.go

```diff
@@ -50,9 +51,27 @@ type RedisClient struct {
 }
 
 // NewRedisClient creates Redis client
-func NewRedisClient(cfg *RedisConfig) *RedisClient {
+func NewRedisClient(cfg *RedisConfig) (*RedisClient, error) {
+	endpoints := strings.Split(cfg.Endpoint, ",")
+	// Handle single configuration endpoint which resolves multiple nodes.
+	if len(endpoints) == 1 {
+		host, port, err := net.SplitHostPort(endpoints[0])
+		if err != nil {
+			return nil, err
+		}
+		addrs, err := net.LookupHost(host)
```

Comment:
This looks to me like a straightforward DNS lookup - why do you call it "Cluster configuration" ?

Reply:
This is a term that AWS uses (only?), the cluster configuration is an A record which resolves all node addresses in a cluster, I used `cluster configuration` to hint that it could be single DNS record pointing to a cluster and reuse the AWS term.
But let me know if you have a suggestion for naming a more generic DNS record pointing to a redis cluster nodes.